extensions/cornerstone/src/commandsModule.ts

```diff
@@ -72,7 +96,129 @@ const commandsModule = ({ servicesManager }) => {
+      options,
+      viewerElement,
+      defaultPointsPosition
+    );
+  }
+
   const actions = {
+    /** Show the specified context menu */
+    showViewerContextMenu: providedOptions => {
+      const viewerElement = _getActiveEnabledElement();
+
+      const options = { ...providedOptions };
+      const { event: evt } = options;
+      const { useSelectedAnnotation, nearbyToolData, menuName } = options;
+
+      if (menuName) {
+        Object.assign(
+          options,
+          customizationService.getModeCustomization(
+            menuName,
+            defaultContextMenu
+          )
+        );
+      }
+
+      if (useSelectedAnnotation && !nearbyToolData) {
+        const firstAnnotationSelected = getFirstAnnotationSelected(
+          viewerElement
+        );
+        // filter by allowed selected tools from config property (if there is any)
+        if (
+          !options.allowedSelectedTools ||
+          options.allowedSelectedTools.includes(
+            firstAnnotationSelected?.metadata?.toolName
+          )
+        ) {
+          options.nearbyToolData = firstAnnotationSelected;
+        } else {
+          return;
+        }
+      }
+
+      // TODO - make the checkProps richer by including the study metadata and display set.
+      options.checkProps = {
+        toolName: options.nearbyToolData?.metadata?.toolName,
+        value: options.nearbyToolData,
+        uid: options.nearbyToolData?.annotationUID,
+        nearbyToolData: options.nearbyToolData,
+      };
+
+      _showViewerContextMenu(viewerElement, options);
+    },
+
+    /** Close any viewer context menus currently displayed */
+    closeViewerContextMenu: () => {
+      contextMenuController.closeViewerContextMenu();
+    },
+
+    getNearbyToolData({ nearbyToolData, element, canvasCoordinates }) {
+      return (
+        nearbyToolData ??
+        cstUtils.getAnnotationNearPoint(element, canvasCoordinates)
+      );
+    },
+
+    // Measurement tool commands:
+    deleteMeasurement: ({ uid }) => {
+      if (uid) {
+        measurementServiceSource.remove(uid);
+      }
+    },
+    setLabel: ({ uid }) => {
+      const measurement = measurementService.getMeasurement(uid);
+
+      callInputDialog(
+        uiDialogService,
+        measurement,
+        (label, actionId) => {
+          if (actionId === 'cancel') {
+            return;
+          }
+
+          const updatedMeasurement = Object.assign({}, measurement, {
+            label,
+          });
+
+          measurementService.update(
+            updatedMeasurement.uid,
+            updatedMeasurement,
+            true
+          );
+        },
+        false
+      );
+    },
+
+    updateMeasurement: props => {
+      const { code, uid, measurementKey = 'finding' } = props;
+      const measurement = measurementService.getMeasurement(uid);
+      const updatedMeasurement = {
+        ...measurement,
+        [measurementKey]: code,
+        label: code.text,
```

Comment:
what happens if code is undefined? error at best case scenario, otherwise we loos the label